### ISSUE TITLE
validation message list contains duplicates

### ISSUE THREAD

**Issue by casualjim** [go-openapi/validate#12]
When validating a spec or schema sometimes the same error occurs many times

For the following spec there is only 1 operation but it renders the same error message 4 times.

``` yaml
swagger: '2.0'
info:
  title: swagger validate out of memory
  version: 0.1.0
paths:
  /foos:
    post:
      operationId: createFoo
      responses:
        200:
          schema:
            $ref: '#/definitions/Foo'

definitions:
  Foo:
    type: object
    properties:
      bar:
        $ref: '#/definitions/Bar'
      baz:
        $ref: '#/definitions/Baz'

  Bar:
    type: string
    enum:
      - a
      - b

  Baz:
    type: array
    items:
      type: string

```

**Comment by fredbi** [comment]
I am on it.

**Comment by casualjim** [comment]
fixed by: go-openapi/validate#44
